be/src/vec/common/hash_table/hash_table.h

```diff
@@ -1111,6 +1118,12 @@ class HashTable : private boost::noncopyable,
     size_t get_buffer_size_in_bytes() const { return grower.buf_size() * sizeof(Cell); }
 
     size_t get_buffer_size_in_cells() const { return grower.buf_size(); }
+    size_t* get_buffer_sizes_in_cells(size_t& num_buckets) const {
+        num_buckets = 1;
```

Comment:
And here.

Reply:
I think, we should not return a pointer and delete outside this method.
Just return a vector, cpp will do return value optimization.... maybe you could test it.
Another method, get_buffer_sizes_in_cells(size_t& num_buckets, std::vector<xxx>* outputvalue)